### PR TITLE
Added improvemnet #10168

### PR DESCRIPTION
**Pull request by amit-webkul**
## Issue Reference
Added improvement #10168

## Description
Added configuration setting for handling censoring product reviewer names on the product page.

Config path: configuration -> catalog -> products -> Review -> Censoring Reviewer Name

## How To Test This?
You can test with enable or disable the configuration and then check on the product view page in the review section. 

## Branch Selection

- [ ] Target Branch: 2.2

### CODE REVIEW

**Comment by ashishkumar-webkul** [comment]
Checked and found that the setting has been added to the configurations and now it is working fine. Please check the Screenshot for reference.

**Screenshot**

https://prnt.sc/U39pb04P2g99
https://prnt.sc/yoyfCa9cTJdJ
https://prnt.sc/ClaFdiG7CW2F

**Comment by devansh-webkul** [comment]
We will give this in master.

File: packages/Webkul/Shop/src/Resources/views/products/view/reviews.blade.php [hunk 1]
```diff
@@ -616,7 +618,19 @@ class="max-h-20 min-w-20 cursor-pointer rounded-xl"
...
+                        const names = this.review.name.split(' ');
+
+                        return names
+                            .map(name => `${name.charAt(0)}${'*'.repeat(name.length - 1)}`)
+                            .join(' ');
+                    } else {
+                        return this.review.name;
+                    }
+                },
             },
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/products/view/reviews.blade.php:634]: Move this logic to backend. There should be no censoring in the frontend.

File: packages/Webkul/Shop/src/Http/Controllers/API/ReviewController.php [hunk 2]
```diff
@@ -35,13 +35,19 @@ public function __construct(
...
             ->reviews()
-            ->Where('status', self::STATUS_APPROVED)
+            ->where('status', self::STATUS_APPROVED)
             ->paginate(8);
 
-        return ProductReviewResource::collection($product);
+        if (core()->getConfigData('catalog.products.review.censoring_reviewer_name')) {
+            $productReviews->getCollection()->each(function ($review) {
+                $review->name = $this->censorReviewerName($review->name);
+            });
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/API/ReviewController.php:47]:
```php
$productReviews->getCollection()->transform(function ($review) {
    $review->name = $this->censorReviewerName($review->name);
    
    return $review;
});
```

File: packages/Webkul/Shop/src/Http/Controllers/API/ReviewController.php [hunk 3]
```diff
@@ -35,13 +35,19 @@ public function __construct(
      */
     public function index(int $id): JsonResource
     {
-        $product = $this->productRepository
+        $productReviews = $this->productRepository
             ->find($id)
             ->reviews()
-            ->Where('status', self::STATUS_APPROVED)
+            ->where('status', self::STATUS_APPROVED)
             ->paginate(8);
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Http/Controllers/API/ReviewController.php:42]:
```php
$product = $this->productRepository->findOrFail($id);
```

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Admin/src/Config/system.php [hunk 4]
```diff
@@ -818,6 +818,11 @@
                 'title'   => 'admin::app.configuration.index.catalog.products.review.allow-customer-review',
                 'type'    => 'boolean',
                 'default' => true,
+            ], [
+                'name'    => 'censoring_reviewer_name',
+                'title'   => 'Censoring Reviewer Name',
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/src/Config/system.php:823]: Translation

Comment (amit-webkul) [reply packages/Webkul/Admin/src/Config/system.php]: Done sir

**Pull request merged**